stellar/deploy-contract.js

```diff
@@ -96,36 +101,83 @@ async function processCommand(options, config, chain) {
     const config = loadConfig(options.env);
     await processor(options, config, getChainConfig(config, options.chainName));
     saveConfig(config, options.env);
 }
 
 function main() {
-    const program = new Command();
-    program.name('deploy-contract').description('Deploy Axelar Soroban contracts on Stellar');
-
-    addBaseOptions(program, { address: true });
-
-    program.addOption(new Option('--initialize', 'initialize the contract'));
-    program.addOption(new Option('--contract-name <contractName>', 'contract name to deploy').makeOptionMandatory(true));
-    program.addOption(new Option('--wasm-path <wasmPath>', 'path to the WASM file').makeOptionMandatory(true));
-    program.addOption(new Option('--nonce <nonce>', 'optional nonce for the signer set'));
-    program.addOption(
-        new Option('--previous-signers-retention <previousSignersRetention>', 'previous signer retention').default(15).argParser(Number),
-    );
-
-    program.addOption(
-        new Option(
-            '--domain-separator <domainSeparator>',
-            'domain separator (pass in the keccak256 hash value OR "offline" meaning that its computed locally)',
-        ).default('offline'),
-    );
-
-    program.action((options) => {
-        mainProcessor(options, processCommand);
-    });
+    // 1st level command
+    const program = new Command('deploy-contract').description('Deploy/Upgrade Soroban contracts on Stellar');
+
+    // 2nd level deploy command
+    const deployCmd = new Command('deploy')
+        .description('Deploy a Soroban contract')
+        .addOption(new Option('--contract-name <contractName>', 'contract name to deploy').makeOptionMandatory(true))
+        .addOption(new Option('--wasm-path <wasmPath>', 'path to the WASM file').makeOptionMandatory(true))
+        .addOption(new Option('--nonce <nonce>', 'optional nonce for the signer set'))
+        .addOption(new Option('--initialize', 'initialize the contract'))
+        .addOption(new Option('--domain-separator <domainSeparator>', 'domain separator (keccak256 hash or "offline")').default('offline'))
+        .addOption(
+            new Option('--previous-signers-retention <previousSignersRetention>', 'previous signer retention')
+                .default(15)
+                .argParser(Number),
+        )
+        .action((options) => {
+            mainProcessor(options, processDeploy);
+        });
+
+    // 2nd level upgrade command
+    const upgradeCmd = new Command('upgrade')
+        .description('Upgrade a Soroban contract')
+        .addOption(new Option('--install', 'install only'))
+        .addOption(new Option('--contract-name <contractName>', 'contract name to deploy'))
+        .addOption(new Option('--wasm-path <wasmPath>', 'path to the WASM file'))
+        .addOption(new Option('--contract-id <contractId>', 'contract id (address) to upgrade'))
+        .addOption(new Option('--new-wasm-hash <newWasmHash>', 'new WASM hash to upgrade'))
```

Comment:
if you merge the install and upgrade step, then you don't need the new wasm hash arg. It can be retrieved dynamically. In general, think about the UX of using the scripts. Their goal is to be more than just a wrapper around the CLI cmd, to simplify operations

Reply:
yup, I agree that merging the install and upgrade step is much simpler. updated.